app/packages/core/src/components/app/Plugins.tsx

```diff
@@ -0,0 +1,170 @@
+import { Engineering } from '@mui/icons-material';
+import {
+  Box,
+  Card,
+  CardActionArea,
+  CardContent,
+  Divider,
+  Grid,
+  MenuItem,
+  Pagination,
+  Select,
+  Stack,
+  TextField,
+  Typography,
+} from '@mui/material';
+import Autocomplete from '@mui/material/Autocomplete';
+import { FunctionComponent, useContext } from 'react';
+import { Link, useNavigate, useSearchParams } from 'react-router-dom';
+
+import { PluginContext } from '../../context/PluginContext';
+
+interface IOptions {
+  clusters: string[];
+  page: number;
+  perPage: number;
+  pluginTypes: string[];
+  search: string;
+}
+
+const optionsFromSearch = (params: URLSearchParams): IOptions => {
+  const clusters = params.get('clusters');
+  const page = params.get('page');
+  const perPage = params.get('perPage');
+  const pluginTypes = params.get('pluginTypes');
+  const search = params.get('search');
+
+  return {
+    clusters: clusters ? clusters.split(',') : [],
+    page: page ? parseInt(page) : 1,
+    perPage: perPage ? parseInt(perPage) : 8,
+    pluginTypes: pluginTypes ? pluginTypes.split(',') : [],
+    search: search ?? '',
+  };
+};
+
+const Plugins: FunctionComponent = () => {
+  const { getAvailableClusters, getAvailablePluginTypes, getPlugin, instances } = useContext(PluginContext);
+  const navigate = useNavigate();
+  const [params] = useSearchParams();
+  const options = optionsFromSearch(params);
```

Comment:
Can you please use the `useQueryState` hook from #468.

Please also have a look at the existing implementation and compare it with your implementation (e.g. debounce the search term).